### PR TITLE
python3Packages.mip: init at 1.13.0

### PR DESCRIPTION
**Pull request by danielbarter**
###### Motivation for this change

adding mip to the python package library

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/mip/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    inherit pname version;
+    sha256 = "0ynczcabpxqwg797wscw1gck4wg574wc2zyp77qsas8jv7xb6b5j";
+  };
+
+  buildInputs = [ setuptools_scm cffi ];
+  doCheck = false;
+
+  meta = with lib; {
+    homepage = "https://github.com/coin-or/python-mip";
+    description = "MIP is a collection of tools for Mixed-Integer Linear programs.";
```
Comment (dotlambda) [inline pkgs/development/python-modules/mip/default.nix:22]:
```suggestion
    description = "Ccollection of tools for Mixed-Integer Linear programs";
```
https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md

File: pkgs/development/python-modules/mip/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+buildPythonPackage rec {
+  pname = "mip";
+  version = "1.13.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0ynczcabpxqwg797wscw1gck4wg574wc2zyp77qsas8jv7xb6b5j";
+  };
+
+  buildInputs = [ setuptools_scm cffi ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/mip/default.nix:17]:
```suggestion
  buildInputs = [ setuptools-scm cffi ];
```

File: pkgs/development/python-modules/mip/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  doCheck = false;
```
Comment (dotlambda) [inline pkgs/development/python-modules/mip/default.nix:18]: Why?

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 128896` at 550444d9 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.mip</li>
    <li>python39Packages.mip</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/mip/default.nix:26:0:
```
   |
26 |     description = "Collection of tools for Mixed-Integer Linear programs";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/mip/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+buildPythonPackage rec {
+  pname = "mip";
+  version = "1.13.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0ynczcabpxqwg797wscw1gck4wg574wc2zyp77qsas8jv7xb6b5j";
+  };
+
+  buildInputs = [ setuptools-scm zlib cffi ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/mip/default.nix:19]: cffi belongs into nativeBuildInputs.

File: pkgs/development/python-modules/mip/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "mip";
+  version = "1.13.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0ynczcabpxqwg797wscw1gck4wg574wc2zyp77qsas8jv7xb6b5j";
+  };
```
Comment (samuela) [inline pkgs/development/python-modules/mip/default.nix:17]: Why not fetchFromGitHub and then run the test suite?

**Comment by SuperSandro2000** [comment]
Closing due to no response from author in a while.

**Pull request closed without merging**